lib/httplib/httplib.go

```diff
@@ -139,15 +139,18 @@ func MakeStdHandlerWithErrorWriter(fn StdHandlerFunc, errWriter ErrorWriter) htt
 
 // WithCSRFProtection ensures that request to unauthenticated API is checked against CSRF attacks
 func WithCSRFProtection(fn HandlerFunc) httprouter.Handle {
-	hanlderFn := MakeHandler(fn)
+	handlerFn := MakeHandler(fn)
 	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
-		err := csrf.VerifyHTTPHeader(r)
-		if err != nil {
-			log.Warningf("unable to validate CSRF token %v", err)
-			trace.WriteError(w, trace.AccessDenied("access denied"))
-			return
+		if r.Method != http.MethodGet && r.Method != http.MethodHead {
```

Comment:
I'm slightly unsure about this, but it shouldn't have any security consequences as long as these methods don't change state, right?

Reply:
It is common practice to exclude "safe" methods for CSRF checks, see e.g. `gorilla/csrf`'s implementation https://github.com/gorilla/csrf/blob/93379db1992f1b2fbdd4ba1b55a1bf0414b0535e/csrf.go#L242-L244 .

OAuth2 flow operates strictly through `GET` (although it is doing mutable stuff), which is not quite REST-y. However OAuth uses its own `state` parameter (which we generate/verify, see the Enterprise counterpart) for those requests, which is really more or less just a CSRF token as well. https://stackoverflow.com/questions/26132066/what-is-the-purpose-of-the-state-parameter-in-oauth-authorization-request

Reply:
Ok, I also thought about the current usages of `WithCSRFProtection`, but I see it was used only for POST/PUT requests.  

Thanks for the detailed explanation :)